packages/o/opensnitch/files/update-config.patch

```diff
@@ -0,0 +1,118 @@
+diff --git a/daemon/Makefile b/daemon/Makefile
+Use /etc/ if files exist in that location
+--- a/daemon/Makefile
++++ b/daemon/Makefile
+@@ -5,15 +5,14 @@ PREFIX?=/usr/local
+ all: opensnitchd
+ 
+ install:
+-	@mkdir -p $(DESTDIR)/etc/opensnitchd/rules
+ 	@install -Dm755 opensnitchd \
+ 		-t $(DESTDIR)$(PREFIX)/bin/
+ 	@install -Dm644 opensnitchd.service \
+-		-t $(DESTDIR)/etc/systemd/system/
++		-t $(DESTDIR)$(PREFIX)/lib64/systemd/system/
+ 	@install -Dm644 default-config.json \
+-		-t $(DESTDIR)/etc/opensnitchd/
++		-t $(DESTDIR)/usr/share/defaults/etc/opensnitchd/
+ 	@install -Dm644 system-fw.json \
+-		-t $(DESTDIR)/etc/opensnitchd/
++		-t $(DESTDIR)/usr/share/defaults/etc/opensnitchd/
+ 	@systemctl daemon-reload
+ 
+ opensnitchd: $(SRC)
+diff --git a/daemon/firewall/config/config.go b/daemon/firewall/config/config.go
+--- a/daemon/firewall/config/config.go
++++ b/daemon/firewall/config/config.go
+@@ -137,7 +137,14 @@ func (c *Config) NewSystemFwConfig(preLoadCb, reLoadCb func()) (*Config, error)
+ 	c.Lock()
+ 	defer c.Unlock()
+ 
+-	c.file = "/etc/opensnitchd/system-fw.json"
++	_, fileErr := os.Stat("/etc/opensnitchd/system-fw.json")
++
++	if fileErr == nil {	
++		c.file = "/etc/opensnitchd/system-fw.json"
++	} else {
++		c.file = "/usr/share/defaults/etc/opensnitchd/system-fw.json"
++	}
++	
+ 	c.monitorExitChan = make(chan bool, 1)
+ 	c.preloadCallback = preLoadCb
+ 	c.reloadCallback = reLoadCb
+diff --git a/daemon/main.go b/daemon/main.go
+--- a/daemon/main.go
++++ b/daemon/main.go
+@@ -62,7 +62,6 @@ var (
+ 	logUTC            = true
+ 	logMicro          = false
+ 	rulesPath         = "/etc/opensnitchd/rules/"
+-	configFile        = "/etc/opensnitchd/default-config.json"
+ 	ebpfModPath       = "" // /usr/lib/opensnitchd/ebpf
+ 	noLiveReload      = false
+ 	queueNum          = 0
+@@ -94,6 +93,18 @@ var (
+ 	resolvMonitor *systemd.ResolvedMonitor
+ )
+ 
++
++func userConfigExists() (string) {
++    _, err := os.Stat("/etc/opensnitchd/default-config.json")
++    if err == nil {
++        return "/etc/opensnitchd/default-config.json"
++    }
++    
++    return "/usr/share/defaults/etc/opensnitchd/default-config.json"
++}
++
++var configFile string = userConfigExists()
++
+ func init() {
+ 	flag.BoolVar(&showVersion, "version", debug, "Show daemon version of this executable and exit.")
+ 	flag.BoolVar(&checkRequirements, "check-requirements", debug, "Check system requirements for incompatibilities.")
+@@ -506,6 +517,8 @@ func main() {
+ 		os.Exit(0)
+ 	}
+ 
++	os.Mkdir("/etc/opensnitchd/rules/", 755)
```

Comment:
You probably want to use `os.MkdirAll()` here; it likely fails because `/etc/opensnitchd` probably doesn't exist yet. The trailing slash in the path also probably needs to be removed. Bonus points for having some basic error handling like printing out a message if it fails, telling why.

Reply:
Is there a way to expand the patch? I do not want to redo the entire patch.